workspaces/marketplace/plugins/marketplace-backend/src/services/MarketplaceCatalogService.ts

```diff
@@ -83,4 +85,69 @@ export class MarketplaceCatalogService implements MarketplaceService {
 
     return result.items as MarketplacePluginList[];
   }
+
+  async getPluginListByName(name: string): Promise<MarketplacePluginList> {
+    const token = await this.auth.getPluginRequestToken({
+      onBehalfOf: await this.auth.getOwnServiceCredentials(),
+      targetPluginId: 'catalog',
+    });
+    const result = await this.catalog.getEntities(
+      {
+        filter: {
+          kind: 'pluginList',
+          'metadata.name': name,
+        },
+      },
+      token,
+    );
+
+    return result.items?.[0] as MarketplacePluginList;
+  }
+
+  async getPluginByName(name: string): Promise<MarketplacePluginEntry> {
+    const token = await this.auth.getPluginRequestToken({
+      onBehalfOf: await this.auth.getOwnServiceCredentials(),
+      targetPluginId: 'catalog',
+    });
+    const result = await this.catalog.getEntities(
```

Comment:
Can we use `getEntityByRef` here?

```suggestion
    const result = await this.catalog.getEntityByRef(
```

Reply:
`getEntityRef` cannot be used as it will just return a single entity. we can use `queryEntities` here and if we pass the limit param as well, then the catalog client returns paginated results!

Reply:
But the function is called `getPluginByName(name: string): Promise<MarketplacePluginEntry>` and returns just one entity? I thought that's exactly for what `getEntityByRef` is?

Reply:
ah okay, makes sense. I thought it was some listing function, my bad. Let me update it.